frontend/packages/operator-lifecycle-manager/src/components/install-plan.tsx

```diff
@@ -237,14 +246,74 @@ export const InstallPlansPage: React.FC<InstallPlansPageProps> = (props) => {
   );
 };
 
+const installPlanPermissionsStateToProps = (state) => ({
+  user: state.UI.get('user'),
+});
+
+const updateUser = (flags: FlagsObject, user: UserKind): string => {
+  if (!flags[FLAGS.OPENSHIFT]) {
+    return authSvc.name();
+  }
+  return user.fullName || user.metadata?.name || '';
+};
+
+const NeedInstallPlanPermissionsWithFlags = ({
+  installPlan,
+  flags,
+  user,
+}: NeedInstallPlanPermissionsWithFlagsProps) => {
+  const [username, setUsername] = React.useState(updateUser(flags, user));
+
+  React.useEffect(() => {
+    setUsername(updateUser(flags, user));
+  }, [flags, user]);
+
+  const { t } = useTranslation();
+
+  const apiGroup = apiGroupForReference(referenceFor(installPlan));
+
+  return (
+    <Alert
+      variant="info"
+      isInline
+      title={t('olm~Missing sufficient privileges for manual InstallPlan approval')}
+    >
+      {t(
+        'olm~User "{{user}}" does not have permissions to update or patch resource InstallPlans in API group "{{apiGroup}}" in the namespace "{{namespace}}."',
+        { user: username, apiGroup, namespace: installPlan.metadata.namespace },
+      )}
+    </Alert>
+  );
+};
+
+export const NeedInstallPlanPermissions = connect(installPlanPermissionsStateToProps)(
+  connectToFlags<NeedInstallPlanPermissionsWithFlagsProps>(FLAGS.OPENSHIFT)(
+    NeedInstallPlanPermissionsWithFlags,
+  ),
+);
+
 export const InstallPlanDetails: React.FC<InstallPlanDetailsProps> = ({ obj }) => {
   const { t } = useTranslation();
   const needsApproval =
     obj.spec.approval === InstallPlanApproval.Manual && obj.spec.approved === false;
 
+  const canUpdateInstallPlans = useAccessReview({
+    group: InstallPlanModel.apiGroup,
+    resource: InstallPlanModel.plural,
+    namespace: obj.metadata.namespace,
+    verb: 'update',
+  });
+
```

Comment:
We don't need both patch and update to modify an InstallPlan resource. Just one or the other. Discussed with @spadgett and he suggested we update to use `k8sPatch` everywhere.
```suggestion
```